test/request.js

```diff
@@ -2327,25 +2311,28 @@ describe('Request', () => {
             server.route({ method: 'POST', path: '/timeout', options: { handler: Hoek.block } });
 
             await server.start();
-            onCleanup.unshift(async () => await server.stop({ timeout: 1 }));
+            onCleanup.unshift(() => server.stop());
 
             const timer = new Hoek.Bench();
             const options = {
-                hostname: '127.0.0.1',
+                hostname: 'localhost',
                 port: server.info.port,
                 path: '/timeout',
                 method: 'POST'
             };
 
-            const res = await new Promise((resolve, reject) => {
+            const req = Http.request(options);
+            onCleanup.unshift(() => req.destroy());
 
-                const req = Http.request(options, resolve);
-                req.once('error', reject);
+            req.on('error', (err) => {
 
-                req.write('\n');
-                onCleanup.unshift(() => req.end());
+                expect(err).to.not.exist();
```

Comment:
This can trigger after the test is done. I would just listen for the `close` using `Events.once()`:

```js
await Events.once(req, 'close');
```

The registers an 'error' listener, and will fail the test if triggered.

Reply:
Good call, this feels a lot better.